crates/kornia-io/src/png.rs

```diff
@@ -68,6 +68,217 @@ pub fn read_image_png_mono16(file_path: impl AsRef<Path>) -> Result<Image<u16, 1
     Ok(Image::new(size.into(), buf_u16)?)
 }
 
+/// Write a grayscale image with a single channel (gray8) to a PNG file.
+///
+/// # Arguments
+///
+/// * `file_path` - The path to save the PNG file.
+/// * `src` - The grayscale image to save.
+///
+/// # Returns
+///
+/// `Ok(())` if the image was successfully written, or an error otherwise.
+///
+/// # Example
+///
+/// ```
+/// use kornia_image::{Image, ImageSize};
+/// use kornia_io::png::write_image_png_gray8;
+///
+/// let image = Image::<u8, 1>::new(
+///     ImageSize {
+///         width: 2,
+///         height: 1,
+///     },
+///     vec![0, 255],
+/// ).unwrap();
+///
+/// write_image_png_gray8("output.png", image).unwrap();
+/// ```
+pub fn write_image_png_gray8(file_path: impl AsRef<Path>, src: &Image<u8, 1>) -> Result<(), IoError> {
+    let file_path = file_path.as_ref();
+    
+    // Create the output file
+    let file = File::create(file_path)?;
+    
+    let width = src.width() as u32;
+    let height = src.height() as u32;
+    
+    // Create PNG encoder
+    let mut encoder = Encoder::new(file, width, height);
+    encoder.set_color(ColorType::Grayscale);
+    encoder.set_depth(png::BitDepth::Eight);
+    
+    let mut writer = encoder.write_header()
+        .map_err(|e| IoError::PngDecodeError(e.to_string()))?;
+    
+    // Write the image data
+    writer.write_image_data(src.as_slice())
+        .map_err(|e| IoError::PngDecodeError(e.to_string()))?;
+    
+    Ok(())
+}
+
+/// Write a RGB image with three channels (rgb8) to a PNG file.
+///
+/// # Arguments
+///
+/// * `file_path` - The path to save the PNG file.
+/// * `src` - The RGB image to save.
+///
+/// # Returns
+///
+/// `Ok(())` if the image was successfully written, or an error otherwise.
+///
+/// # Example
+///
+/// ```
+/// use kornia_image::{Image, ImageSize};
+/// use kornia_io::png::write_image_png_rgb8;
+///
+/// let image = Image::<u8, 3>::new(
+///     ImageSize {
+///         width: 2,
+///         height: 1,
+///     },
+///     vec![255, 0, 0, 0, 255, 0],
+/// ).unwrap();
+///
+/// write_image_png_rgb8("output.png", image).unwrap();
+/// ```
+pub fn write_image_png_rgb8(file_path: impl AsRef<Path>, src: Image<u8, 3>) -> Result<(), IoError> {
```

Comment:
```suggestion
pub fn write_image_png_rgb8(file_path: impl AsRef<Path>, src: &Image<u8, 3>) -> Result<(), IoError> {
```
same for all the other functionality

Reply:
done!